torchgeo/datasets/stacapidataset.py

```diff
@@ -0,0 +1,321 @@
+            minx, miny, maxx, maxy = item.bbox
+
+            transformer = Transformer.from_crs(src_crs.to_epsg(), crs.to_epsg())
+            (minx, maxx), (miny, maxy) = transformer.transform(
+                [minx, maxx], [miny, maxy]
+            )
+            mint = 0
+            maxt = sys.maxsize
+            coords = (minx, maxx, miny, maxy, mint, maxt)
+            self.index.insert(i, coords, item)
+
+        self._crs = crs
+        self.res = 10
+        self.transforms = transforms
+        self.items = items
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image/mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        items = [hit.object for hit in hits]
+
+        if not items:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        bounds = (query.minx, query.miny, query.maxx, query.maxy)
+
+        raster_list = []
+        for item in items:
+            raster_list.append(self._snap_to_single_raster(item, bounds))
+
+        # merge single rasters
+        data = self._merge_rasters(raster_list)
+
+        # if only single time step then squeeze out time dimenstion
+        image = data.squeeze(0)
+
+        # suggested #
+        signed_items = [pc.sign(item).to_dict() for item in items]
+
+        stack = stackstac.stack(
+            signed_items,
+            assets=self.bands,
+            resolution=self.res,
+            epsg=self._crs.to_epsg(),
+        )
+
+        aoi = stack.loc[
+            ..., query.maxy : query.miny, query.minx : query.maxx  # type: ignore[misc]
+        ]
+
+        suggested_data = aoi.compute(scheduler="single-threaded")
+        suggested_image = suggested_data.data
+        # end suggested #
+
+        assert suggested_image.shape == image.shape
+
+        key = "image" if self.is_image else "mask"
+        sample = {key: image, "crs": self.crs, "bbox": query}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _snap_to_single_raster(
+        self, item: Item, bounds: Tuple[float, ...]
+    ) -> DataArray:
+        """Load and merge one or multiple individual bands to one raster.
+
+        Args:
+            item: one search item from cataloge
+            bounds: (minx, maxx, miny, maxy) coordinates to index
+
+        Returns:
+            computed data array from stac
+        """
+        signed_items = pc.sign(item).to_dict()
+
+        aoi = stackstac.stack(
+            signed_items,
+            assets=self.bands,
+            bounds_latlon=bounds,
```

Comment:
Bounds are not in EPSG:4326 now but the STAC dataset's native CRS (i.e. some UTM projection).
```suggestion
            bounds=bounds,
```